repos/system_upgrade/el7toel8/actors/satelliteupgradefacts/actor.py

```diff
@@ -0,0 +1,133 @@
+import os
+import textwrap
+
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import run
+from leapp.models import InstalledRPM, Module, RepositoriesSetupTasks, RpmTransactionTasks
+from leapp.tags import FactsPhaseTag, IPUWorkflowTag
+
+
+class SatelliteUpgradeFacts(Actor):
+    """
+    No documentation has been provided for the satellite_upgrade_facts actor.
+    """
+
+    name = 'satellite_upgrade_facts'
+    consumes = (InstalledRPM, )
+    produces = (RepositoriesSetupTasks, RpmTransactionTasks, reporting.Report)
+    tags = (IPUWorkflowTag, FactsPhaseTag)
+
+    def process(self):
+        has_foreman = has_package(InstalledRPM, 'foreman') or has_package(InstalledRPM, 'foreman-proxy')
+        if not has_foreman:
+            return
+
+        local_postgresql = has_package(InstalledRPM, 'rh-postgresql12-postgresql-server')
+        postgresql_contrib = has_package(InstalledRPM, 'rh-postgresql12-postgresql-contrib')
+        postgresql_evr = has_package(InstalledRPM, 'rh-postgresql12-postgresql-evr')
+        # remove packages we don't need anymore
+        self.produce(RpmTransactionTasks(
+            to_remove=['tfm-runtime', 'tfm-pulpcore-runtime', 'rh-redis5-runtime', 'rh-ruby27-runtime', 'rh-python38-runtime'],
+            # you'll need to define the repos in leapp_upgrade_repositories.repo
+            # until this is on the CDN
+            to_install=['rubygem-foreman_maintain', 'foreman-installer-katello'],
+            modules_to_enable=[Module(name='ruby', stream='2.7'), Module(name='python38', stream='3.8')]
+            )
+        )
+
+        if has_package(InstalledRPM, 'katello'):
```

Comment:
Technically this is only needed for candlepin, not katello itself, right?

Reply:
I think this is just used as an detection mechanism AFAICS

Reply:
It depends on how you look at this. My intention, with these line was "ensure katello and katello-selinux are installed", which only is needed if "katello" was already installed, but has the dependency (via candlepin) on the PKI modules.

Reply:
I should also clarify that I'd like to get rid of the need to install the `katello` package, but perhaps I'm being premature here. The dependency chain is katello-selinux -> candlepin-selinux -> candlepin -> some package from the pki-core module. Perhaps that should be included in a code comment and that's sufficient?